src/email.rs

```diff
@@ -0,0 +1,169 @@
+use std::ffi::CString;
+use std::{mem, ptr};
+
+use crate::util::Binding;
+use crate::{raw, Buf, Commit, DiffFindOptions, DiffOptions, Error, IntoCString};
+use crate::{Diff, Oid, Signature};
+
+/// A structure to represent patch in mbox format for sending via email
+pub struct Email {
+    /// Buffer to store the e-mail patch in
+    pub buf: Buf,
+}
+
+/// Options for controlling the formatting of the generated e-mail.
+pub struct EmailCreateOptions {
+    /// Options to use when creating diffs
+    pub diff_options: DiffOptions,
+    /// Options for finding similarities within diffs
+    pub diff_find_options: DiffFindOptions,
```

Comment:
Instead of making these `pub` could method accessors be provided?

Reply:
Done.